merge_duplicate_contacts/wizard/merge_contact.py

```diff
@@ -153,66 +153,42 @@ def _action_new_next_screen(self):
-            "'jpberlin.de','kabelmail.de','laposte.net','lavabit.com','librem.one',"
-            "'live.co.uk','live.com','live.de','live.fr','live.nl','mac.com','mail.de',"
-            "'mailbox.org','mailfence.com','me.com','meineinkauf.ch','msn.com',"
-            "'mykolab.ch','mykolab.com','netcologne.de','online.de','onlinehome.de',"
-            "'orange.fr','outlook.com','outlook.de','outlook.es','outlook.fr',"
-            "'posteo.at','posteo.ch','posteo.co.uk','posteo.de','posteo.eu',"
-            "'posteo.lu','posteo.net','posteo.org','proton.me','protommail.com',"
-            "'protonmail.ch','pt.lu','riseup.net','runbox.com','secure.mailbox.org',"
-            "'startmail.com','system.li','temp.mailbox.org','t-online.de',"
-            "'tuta.io','tutamail.com','tutanota.com','tutanota.de','vodafone.de',"
-            "'vodafonemail.de','wanadoo.fr','web.de','xs4all.nl','yahoo.ca','yahoo.co.uk',"
-            "'yahoo.com','yahoo.de','yahoo.es','yahoo.fr','yahoo.it')"
+        email_domains_config_value = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("merge_duplicate_contacts.email_domains")
         )
 
+        if email_domains_config_value:
+            email_domains = email_domains_config_value.split(",")
+        else:
+            email_domains = []
+
         # Set up parts of the querries based on selected fields here
         # Match case would look better here but isn't a feature in our python version yet
+        sql_fields = []
+        where_querries = []
         for field in fields:
-            if field == "email":
-                group_by_querries.append("email")
-
-            if field == "name":
-                group_by_querries.append("name")
+            if field in ["email", "name"]:
+                sql_fields.append("lower(%s)" % field)
 
-            if field == "phone":
-                group_by_querries.append("phone")
+            elif field in ["vat"]:
+                sql_fields.append("replace(%s, ' ', '')" % field)
 
-            if field == "mobile":
-                group_by_querries.append("mobile")
+            else:
+                sql_fields.append(field)
 
-            if field == "is_company":
-                group_by_querries.append("is_company")
-
-            if field == "vat":
-                group_by_querries.append("vat")
-
-            if field == "parent_id":
-                group_by_querries.append("parent_id")
-
-            if field == "domain_email":
-                where_querries.append(
-                    "substring(email from '@(.*)$') not in %s"
-                    % commercial_email_domains
-                )
+        if email_domains and self.filter_domain_email:
+            formatted_domains = "'" + "', '".join(email_domains) + "'"
+            where_querries.append(
+                f"substring(email from '@(.*)$') not in ({formatted_domains})"
```

Comment:
This is how SQL injections are built. If I add some `' --` to the config I can get out and execute random queries.

Please use:
```
clause = self.env.cr.mogrify("substring(email from '@(.*)$') not in %s", [email_domains]).decode()
```

Reply:
I had to change it to:
```python
clause = self.env.cr.mogrify(
                "substring(email from '@(.*)$') not in %s", (tuple(email_domains),)
            ).decode()
```
otherwise it failed with:
```python
Traceback (most recent call last):
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/odoo/odoo/parts/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
psycopg2.errors.SyntaxError: syntax error at or near "ARRAY"
LINE 1: ...tner  WHERE substring(email from '@(.*)$') not in ARRAY['aik...
```

Reply:
fair enough